File: java-vertx/src/main/java/org/openapitools/server/api/model/BasicBotBase.java
package org.openapitools.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.ArrayList;
import java.util.List;
import org.openapitools.server.api.model.OneOfobjectobject;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class BasicBotBase   {
  
  private Integer userId;
  private String fullName;
  private String apiKey;
  private String defaultSendingStream;
  private String defaultEventsRegisterStream;
  private Boolean defaultAllPublicStreams;
  private String avatarUrl;
  private Integer ownerId;
  private List<OneOfobjectobject> services = new ArrayList<>();

  public BasicBotBase () {

  }

  public BasicBotBase (Integer userId, String fullName, String apiKey, String defaultSendingStream, String defaultEventsRegisterStream, Boolean defaultAllPublicStreams, String avatarUrl, Integer ownerId, List<OneOfobjectobject> services) {
    this.userId = userId;
    this.fullName = fullName;
    this.apiKey = apiKey;
    this.defaultSendingStream = defaultSendingStream;
    this.defaultEventsRegisterStream = defaultEventsRegisterStream;
    this.defaultAllPublicStreams = defaultAllPublicStreams;
    this.avatarUrl = avatarUrl;
    this.ownerId = ownerId;
    this.services = services;
  }

    
  @JsonProperty("user_id")
  public Integer getUserId() {
    return userId;
  }
  public void setUserId(Integer userId) {
    this.userId = userId;
  }

    
  @JsonProperty("full_name")
  public String getFullName() {
    return fullName;
  }
  public void setFullName(String fullName) {
    this.fullName = fullName;
  }

    
  @JsonProperty("api_key")
  public String getApiKey() {
    return apiKey;
  }
  public void setApiKey(String apiKey) {
    this.apiKey = apiKey;
  }

    
  @JsonProperty("default_sending_stream")
  public String getDefaultSendingStream() {
    return defaultSendingStream;
  }
  public void setDefaultSendingStream(String defaultSendingStream) {
    this.defaultSendingStream = defaultSendingStream;
  }

    
  @JsonProperty("default_events_register_stream")
  public String getDefaultEventsRegisterStream() {
    return defaultEventsRegisterStream;
  }
  public void setDefaultEventsRegisterStream(String defaultEventsRegisterStream) {
    this.defaultEventsRegisterStream = defaultEventsRegisterStream;
  }

    
  @JsonProperty("default_all_public_streams")
  public Boolean getDefaultAllPublicStreams() {
    return defaultAllPublicStreams;
  }
  public void setDefaultAllPublicStreams(Boolean defaultAllPublicStreams) {
    this.defaultAllPublicStreams = defaultAllPublicStreams;
  }

    
  @JsonProperty("avatar_url")
  public String getAvatarUrl() {
    return avatarUrl;
  }
  public void setAvatarUrl(String avatarUrl) {
    this.avatarUrl = avatarUrl;
  }

    
  @JsonProperty("owner_id")
  public Integer getOwnerId() {
    return ownerId;
  }
  public void setOwnerId(Integer ownerId) {
    this.ownerId = ownerId;
  }

    
  @JsonProperty("services")
  public List<OneOfobjectobject> getServices() {
    return services;
  }
  public void setServices(List<OneOfobjectobject> services) {
    this.services = services;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    BasicBotBase basicBotBase = (BasicBotBase) o;
    return Objects.equals(userId, basicBotBase.userId) &&
        Objects.equals(fullName, basicBotBase.fullName) &&
        Objects.equals(apiKey, basicBotBase.apiKey) &&
        Objects.equals(defaultSendingStream, basicBotBase.defaultSendingStream) &&
        Objects.equals(defaultEventsRegisterStream, basicBotBase.defaultEventsRegisterStream) &&
        Objects.equals(defaultAllPublicStreams, basicBotBase.defaultAllPublicStreams) &&
        Objects.equals(avatarUrl, basicBotBase.avatarUrl) &&
        Objects.equals(ownerId, basicBotBase.ownerId) &&
        Objects.equals(services, basicBotBase.services);
  }

  @Override
  public int hashCode() {
    return Objects.hash(userId, fullName, apiKey, defaultSendingStream, defaultEventsRegisterStream, defaultAllPublicStreams, avatarUrl, ownerId, services);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class BasicBotBase {\n");
    
    sb.append("    userId: ").append(toIndentedString(userId)).append("\n");
    sb.append("    fullName: ").append(toIndentedString(fullName)).append("\n");
    sb.append("    apiKey: ").append(toIndentedString(apiKey)).append("\n");
    sb.append("    defaultSendingStream: ").append(toIndentedString(defaultSendingStream)).append("\n");
    sb.append("    defaultEventsRegisterStream: ").append(toIndentedString(defaultEventsRegisterStream)).append("\n");
    sb.append("    defaultAllPublicStreams: ").append(toIndentedString(defaultAllPublicStreams)).append("\n");
    sb.append("    avatarUrl: ").append(toIndentedString(avatarUrl)).append("\n");
    sb.append("    ownerId: ").append(toIndentedString(ownerId)).append("\n");
    sb.append("    services: ").append(toIndentedString(services)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: cpp-tiny/lib/Models/BasicBotBase.cpp


#include "BasicBotBase.h"

using namespace Tiny;

BasicBotBase::BasicBotBase()
{
	user_id = int(0);
	full_name = std::string();
	api_key = std::string();
	default_sending_stream = std::string();
	default_events_register_stream = std::string();
	default_all_public_streams = bool(false);
	avatar_url = std::string();
	owner_id = int(0);
	services = std::list<OneOf&lt;object,object&gt;>();
}

BasicBotBase::BasicBotBase(std::string jsonString)
{
	this->fromJson(jsonString);
}

BasicBotBase::~BasicBotBase()
{

}

void
BasicBotBase::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *user_idKey = "user_id";
    
    if(object.has_key(user_idKey)) 
    {
        bourne::json value = object[user_idKey];


        
        jsonToValue(&user_id, value, "int");


    }

    const char *full_nameKey = "full_name";
    
    if(object.has_key(full_nameKey)) 
    {
        bourne::json value = object[full_nameKey];


        
        jsonToValue(&full_name, value, "std::string");


    }

    const char *api_keyKey = "api_key";
    
    if(object.has_key(api_keyKey)) 
    {
        bourne::json value = object[api_keyKey];


        
        jsonToValue(&api_key, value, "std::string");


    }

    const char *default_sending_streamKey = "default_sending_stream";
    
    if(object.has_key(default_sending_streamKey)) 
    {
        bourne::json value = object[default_sending_streamKey];


        
        jsonToValue(&default_sending_stream, value, "std::string");


    }

    const char *default_events_register_streamKey = "default_events_register_stream";
    
    if(object.has_key(default_events_register_streamKey)) 
    {
        bourne::json value = object[default_events_register_streamKey];


        
        jsonToValue(&default_events_register_stream, value, "std::string");


    }

    const char *default_all_public_streamsKey = "default_all_public_streams";
    
    if(object.has_key(default_all_public_streamsKey)) 
    {
        bourne::json value = object[default_all_public_streamsKey];


        
        jsonToValue(&default_all_public_streams, value, "bool");


    }

    const char *avatar_urlKey = "avatar_url";
    
    if(object.has_key(avatar_urlKey)) 
    {
        bourne::json value = object[avatar_urlKey];


        
        jsonToValue(&avatar_url, value, "std::string");


    }

    const char *owner_idKey = "owner_id";
    
    if(object.has_key(owner_idKey)) 
    {
        bourne::json value = object[owner_idKey];


        
        jsonToValue(&owner_id, value, "int");


    }

    const char *servicesKey = "services";
    
    if(object.has_key(servicesKey)) 
    {
        bourne::json value = object[servicesKey];


        std::list<OneOf&lt;object,object&gt;> services_list;
        OneOf&lt;object,object&gt; element;
        for(auto& var : value.array_range())
        {

            
            element.fromJson(var.dump());
            
            services_list.push_back(element);
        }
        services = services_list;


    }


}

bourne::json
BasicBotBase::toJson()
{
    bourne::json object = bourne::json::object();

    



    object["user_id"] = getUserId();


    



    object["full_name"] = getFullName();


    



    object["api_key"] = getApiKey();


    



    object["default_sending_stream"] = getDefaultSendingStream();


    



    object["default_events_register_stream"] = getDefaultEventsRegisterStream();


    



    object["default_all_public_streams"] = isDefaultAllPublicStreams();


    



    object["avatar_url"] = getAvatarUrl();


    



    object["owner_id"] = getOwnerId();


    
    

    std::list<OneOf&lt;object,object&gt;> services_list = getServices();
    bourne::json services_arr = bourne::json::array();

    for(auto& var : services_list)
    {
        OneOf&lt;object,object&gt; obj = var;
        services_arr.append(obj.toJson());
    }
    object["services"] = services_arr;




    return object;

}

int
BasicBotBase::getUserId()
{
	return user_id;
}

void
BasicBotBase::setUserId(int  user_id)
{
	this->user_id = user_id;
}

std::string
BasicBotBase::getFullName()
{
	return full_name;
}

void
BasicBotBase::setFullName(std::string  full_name)
{
	this->full_name = full_name;
}

std::string
BasicBotBase::getApiKey()
{
	return api_key;
}

void
BasicBotBase::setApiKey(std::string  api_key)
{
	this->api_key = api_key;
}

std::string
BasicBotBase::getDefaultSendingStream()
{
	return default_sending_stream;
}

void
BasicBotBase::setDefaultSendingStream(std::string  default_sending_stream)
{
	this->default_sending_stream = default_sending_stream;
}

std::string
BasicBotBase::getDefaultEventsRegisterStream()
{
	return default_events_register_stream;
}

void
BasicBotBase::setDefaultEventsRegisterStream(std::string  default_events_register_stream)
{
	this->default_events_register_stream = default_events_register_stream;
}

bool
BasicBotBase::isDefaultAllPublicStreams()
{
	return default_all_public_streams;
}

void
BasicBotBase::setDefaultAllPublicStreams(bool  default_all_public_streams)
{
	this->default_all_public_streams = default_all_public_streams;
}

std::string
BasicBotBase::getAvatarUrl()
{
	return avatar_url;
}

void
BasicBotBase::setAvatarUrl(std::string  avatar_url)
{
	this->avatar_url = avatar_url;
}

int
BasicBotBase::getOwnerId()
{
	return owner_id;
}

void
BasicBotBase::setOwnerId(int  owner_id)
{
	this->owner_id = owner_id;
}

std::list<OneOf&lt;object,object&gt;>
BasicBotBase::getServices()
{
	return services;
}

void
BasicBotBase::setServices(std::list <OneOf&lt;object,object&gt;> services)
{
	this->services = services;
}



